design_drafts/components_architecture_and_urdf_examples.md

```diff
@@ -0,0 +1,424 @@
+    </joint>
+  </ros2_control>
+  <ros2_control name="2DOF_System_Robot_ForceTorqueSensor" type="sensor">
+    <hardware>
+      <classType>ros2_control_demo_hardware/2D_Sensor_Force_Torque</classType>
+      <param name="example_param_read_for_sec">2</param>
+    </hardware>
+    <sensor name="tcp_fts_sensor">
+      <classType>ros2_control_components/ForceTorqueSensor</classType>
+      <param name="frame_id">kuka_tcp</frame_id>
+      <param name="lower_limits">-100</param>
+      <param name="upper_limits">100</param>
+    </sensor>
+  </ros2_control>
+```
+
+#### 6. Modular Robots with separate communication to each actuator
+  * the communication is done on actuator level using proprietary or standardized API (e.g., canopen_402, Modbus, RS232, RS485)
+  * Data for all actuators is exchanged separately from each other
+  * Examples: Mara, Arduino-based-robots
+
+```xml
+  <ros2_control name="2DOF_Modular_Robot_joint1"  type="actuator">
+    <hardware>
+      <classType>ros2_control_demo_hardware/Position_Actuator_Hadware</classType>
+      <param name="example_param_write_for_sec">1.23</param>
+      <param name="example_param_read_for_sec">3</param>
+    </hardware>
+    <joint name="joint1">
+      <classType>ros2_control_components/PositionJoint</classType>
+      <param name="min_values">-1</param>
+      <param name="max_values">1</param>
+    </joint>
+  </ros2_control>
+  <ros2_control name="2DOF_Modular_Robot_joint2"  type="actuator">
+    <hardware>
+      <classType>ros2_control_demo_hardware/Position_Actuator_Hadware</classType>
+      <param name="example_param_write_for_sec">1.23</param>
+      <param name="example_param_read_for_sec">3</param>
+    </hardware>
+    <joint name="joint2">
+      <classType>ros2_control_components/PositionJoint</classType>
+      <param name="min_values">-1</param>
+      <param name="max_values">1</param>
+    </joint>
+  </ros2_control>
+```
+
+#### 7. Modular Robots with actuators not providing states and additional sensors (simple Transmissions)
+  * the communication is done on actuator level using proprietary or standardized API (e.g., canopen_402, modbus, RS232, RS485)
+  * Data for all actuators and sensors is exchanged separately from each other
+  * Examples: Arduino-based-robots, custom robots
+
+```xml
+  <ros2_control name="2DOF_Modular_Robot_joint1"  type="actuator">
+    <hardware>
+      <classType>ros2_control_demo_hardware/Velocity_Actuator_Hadware</classType>
+      <param name="example_param_write_for_sec">1.23</param>
+      <param name="example_param_read_for_sec">3</param>
+    </hardware>
+    <joint name="joint1">
+      <classType>ros2_control_components/VelocityJoint</classType>
+      <commandInterfaceType>position</commandInterfaceType>
+      <param name="min_values">-1</param>
+      <param name="max_values">1</param>
+    </joint>
+    <transmission name="transmission1">
+      <type>transmission_interface/SimpleTansmission</type>
+      <joint name="joint1">
```

Comment:
I could see how "joint" becomes quite confusing here.

Reply:
can we not just directly pass in all listed joints within the `ros2_control` tag to the transmission plugin? I mean it doesn't make much sense to specify "joint5" here if that joint is never being used within that component.

Reply:
This is true. For the actuators we can do this implicitly.